File: client/csharp/Trash.cs
using System;
using System.Runtime.InteropServices;


///This is a Trash object
public class Trash: Mappable
{

  public Trash()
  {
  }

  public Trash(IntPtr p)
  {
    ptr = p;
    ID = Client.trashGetId(ptr);
    iteration = BaseAI.iteration;
  }

  public override bool validify()
  {
    if(iteration == BaseAI.iteration) return true;
    for(int i = 0; i < BaseAI.trashs.Length; i++)
    {
      if(BaseAI.trashs[i].ID == ID)
      {
        ptr = BaseAI.trashs[i].ptr;
        iteration = BaseAI.iteration;
        return true;
      }
    }
    throw new ExistentialError();
  }

    //commands


    //getters


  ///Unique Identifier
  public new int Id
  {
    get
    {
      validify();
      int value = Client.trashGetId(ptr);
      return value;
    }
  }

  ///X position of the object
  public new int X
  {
    get
    {
      validify();
      int value = Client.trashGetX(ptr);
      return value;
    }
  }

  ///Y position of the object
  public new int Y
  {
    get
    {
      validify();
      int value = Client.trashGetY(ptr);
      return value;
    }
  }

  ///The weight of the trash
  public int Weight
  {
    get
    {
      validify();
      int value = Client.trashGetWeight(ptr);
      return value;
    }
  }

}

File: client/java/Trash.java
import com.sun.jna.Pointer;

///This is a Trash object
class Trash extends Mappable
{
  public Trash(Pointer p)
  {
    super(p);
  }
  boolean validify()
  {
    if(iteration == BaseAI.iteration) return true;
    for(int i = 0; i < BaseAI.trashs.length; i++)
    {
      if(BaseAI.trashs[i].ID == ID)
      {
        ptr = BaseAI.trashs[i].ptr;
        iteration = BaseAI.iteration;
        return true;
      }
    }
    throw new ExistentialError();
  }

    //commands


    //getters

  ///Unique Identifier
  public int getId()
  {
    validify();
    return Client.INSTANCE.trashGetId(ptr);
  }
  ///X position of the object
  public int getX()
  {
    validify();
    return Client.INSTANCE.trashGetX(ptr);
  }
  ///Y position of the object
  public int getY()
  {
    validify();
    return Client.INSTANCE.trashGetY(ptr);
  }
  ///The weight of the trash
  public int getWeight()
  {
    validify();
    return Client.INSTANCE.trashGetWeight(ptr);
  }

}
